From 24073abf0e039d10d840f8039330489e6dcbb78e Mon Sep 17 00:00:00 2001
From: Zelan Zou <zelan.zou@nxp.com>
Date: Mon, 28 Jul 2026 00:00:00 +0000
Subject: [PATCH] arm: tools: gen-mach-types: use basename to avoid embedding
 absolute build paths

The awk built-in variable FILENAME contains the full absolute path of
the input file. When building with Yocto/OE, this embeds the host
TMPDIR path (e.g. /opt/.../tmp/work-shared/.../arch/arm/tools/mach-types)
into the generated mach-types.h header comment, which triggers the
buildpaths QA error.

Use only the basename of FILENAME to keep the generated header
reproducible and free of host-specific paths.

Upstream-Status: Pending
Signed-off-by: Zelan Zou <zelan.zou@nxp.com>
---
 arch/arm/tools/gen-mach-types | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/tools/gen-mach-types b/arch/arm/tools/gen-mach-types
index cbe1c33bb871..efcbb45c426f 100644
--- a/arch/arm/tools/gen-mach-types
+++ b/arch/arm/tools/gen-mach-types
@@ -23,8 +23,9 @@ NF == 3 {
 
 
 END	{
+	  n = split(FILENAME, _parts, "/");
 	  printf("/*\n");
-	  printf(" * This was automagically generated from %s!\n", FILENAME);
+	  printf(" * This was automagically generated from %s!\n", _parts[n]);
 	  printf(" * Do NOT edit\n");
 	  printf(" */\n\n");
 	  printf("#ifndef __ASM_ARM_MACH_TYPE_H\n");
